### PR TITLE
Make parser arg optional, rename main function for github-url-parser

### PR DESCRIPTION
**Pull request by humphd**
I need to make a few tweaks to the API.

### CODE REVIEW

**Review by DukeManh — APPROVED** [review]
That e2e test again.

**Comment by humphd** [comment]
> That e2e test again.

Filed https://github.com/Seneca-CDOT/telescope/issues/3504.  We need to solve this.

**Comment by humphd** [comment]
I'm doing a test to see if I can get more details about this failure.  I'll undo these changes.

**Comment by humphd** [comment]
Can I get some reviews on this so I can build on it?  We're all clear on CI

File: src/github-url-parser/test/index.test.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-const { githubUrlParser } = require('../dist/main');
+const { parseGitHubUrls } = require('../dist/main');
 const { parser } = require('./util');
 
 describe('githubUrlParser', () => {
```
Comment (menghif) [inline src/github-url-parser/test/index.test.js:4]: `parseGitHubUrls` here too.

**Review by menghif — CHANGES_REQUESTED** [review]

File: src/github-url-parser/test/index.test.js (hunk 1 above)
Comment (humphd) [reply src/github-url-parser/test/index.test.js]: Fixed.

**Review by menghif — APPROVED** [review]

**Review by DukeManh — APPROVED** [review]

**Pull request merged**